query/src/lib.rs

```diff
@@ -474,13 +474,30 @@ pub struct Predicate {
     pub args: Vec<FnArg>,
 }
 
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum UnifyVars {
+    Implicit,
```

Comment:
Worth a comment on the difference here.  Am I correct that `Implicit` is equivalent to `Explicit` with the complete set of variables occurring in all (equivalently, each) clause?